### PR TITLE
Merge release 1.2.2 into master

### PR DESCRIPTION
**Pull request by bhennion**
@xournalpp/core Please have a look and check that nothing got lost in conflicts.

### CODE REVIEW

**Comment by Febbe** [comment]
@bhennion have reviewed around about 50% and it looks ok on the first look. Can you tell me specific files, where you got merge conflicts?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,27 @@
 # Changelog
 
-## 1.2.1+dev (Unreleased)
+## 1.2.3+dev (Unreleased)
```
Comment (rolandlo) [inline CHANGELOG.md:3]:
```suggestion
## 1.2.2+dev (Unreleased)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,10 +3,15 @@ cmake_policy(VERSION ${CMAKE_VERSION})
...
+if(NOT CMAKE_BUILD_TYPE)
+    # set default build type to RelWithDebInfo, we never supported multi config generators, therefore this is fine
+    set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel." FORCE)
+endif()
+
 ## Project related Variables
 ## Also update changelog in debian folder!
 project(xournalpp
-        VERSION 1.2.1
+        VERSION 1.2.3
```
Comment (rolandlo) [inline CMakeLists.txt:14]:
```suggestion
        VERSION 1.2.2
```

File: debian/changelog [hunk 3]
```diff
@@ -1,8 +1,17 @@
-xournalpp (1.2.1+dev-1) UNRELEASED; urgency=medium
+xournalpp (1.2.3+dev-1) UNRELEASED; urgency=medium
```
Comment (rolandlo) [inline debian/changelog:1]:
```suggestion
xournalpp (1.2.2+dev-1) UNRELEASED; urgency=medium
```

File: mac-setup/Info.plist [hunk 4]
```diff
@@ -16,9 +16,9 @@
     <string>Xournal++</string>
 
     <key>CFBundleShortVersionString</key>
-    <string>1.2.1+dev</string>
+    <string>1.2.3+dev</string>
```
Comment (rolandlo) [inline mac-setup/Info.plist:19]:
```suggestion
    <string>1.2.2+dev</string>
```

File: mac-setup/Info.plist [hunk 5]
```diff
@@ -16,9 +16,9 @@
...
     <key>CFBundleVersion</key>
-    <string>1.2.1+dev.0</string>
+    <string>1.2.3+dev.0</string>
```
Comment (rolandlo) [inline mac-setup/Info.plist:21]:
```suggestion
    <string>1.2.2+dev.0</string>
```

**Review by rolandlo — COMMENTED** [review]
The version numbers look wrong. Otherwise the code looks good. I will build and test, if I can see a problem.

**Comment by rolandlo** [comment]
I think ee03a0e7c3491121bb17c48d0728fb181f5fa8c8 shouldn't be included.
See https://github.com/xournalpp/xournalpp/pull/5090#issuecomment-1698591977 and https://github.com/xournalpp/xournalpp/pull/5090#issuecomment-1695013317.

**Comment by rolandlo** [comment]
> > I think [ee03a0e](https://github.com/xournalpp/xournalpp/commit/ee03a0e7c3491121bb17c48d0728fb181f5fa8c8) shouldn't be included. See [#5090 (comment)](https://github.com/xournalpp/xournalpp/pull/5090#issuecomment-1698591977) and [#5090 (comment)](https://github.com/xournalpp/xournalpp/pull/5090#issuecomment-1695013317).
> 
> I don't quite understand this. Is there a reason for that or is it just a force of habits?

We want to merge back release 1.2.2 into master. So the last commit merged should be the release commit, not the bump to version 1.2.3~dev, which doesn't make any sense on master, where the logical step is from 1.2.2 to 1.2.2+dev (which is lower than 1.2.3~dev).

**Comment by rolandlo** [comment]
Well, if we merge back the release branch into master in an earlier stage than after a release, there is no version bump commit and no need to drop any commit.

**Comment by rolandlo** [comment]
Hmm, I see the issue now.

**Comment by rolandlo** [comment]
Ok, makes sense.

**Comment by Febbe** [comment]
I think, the reason, why we never included the bump is, that we can't know, if we ever push a commit after it. But as long as the version on the master is always any successor (+dev) of the latest release this should be fine. The bump itself must not prevent us from merging unfinished Bugfix releases.

**Pull request merged**